backend/pkg/boards/events.go

```diff
@@ -0,0 +1,92 @@
+package boards
+
+import (
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/abstraction"
+)
+
+type AckNotification struct {
+	ID abstraction.BoardEvent // AckId
+}
+
+func (ack AckNotification) Event() abstraction.BoardEvent {
+	return ack.ID
+}
+
+type DownloadEvent struct {
+	EventID abstraction.BoardEvent // DownloadId
+	BoardID abstraction.BoardId
+}
+
+func (download DownloadEvent) Topic() abstraction.BrokerTopic {
+	return abstraction.BrokerTopic(download.EventID)
+}
+
+func (download DownloadEvent) Event() abstraction.BoardEvent {
+	return download.EventID
+}
+
+type UploadEvent struct {
+	EventID abstraction.BoardEvent // UploadId
+	BoardID abstraction.BoardId
+	Data    []byte
+}
+
+func (upload UploadEvent) Topic() abstraction.BrokerTopic {
+	return abstraction.BrokerTopic(upload.EventID)
+}
+
+func (upload UploadEvent) Event() abstraction.BoardEvent {
+	return upload.EventID
+}
+
+type BoardPush struct {
+	ID   abstraction.BrokerTopic // BlcuOrderId
+	Data int64
+}
+
+func (boardPush BoardPush) Topic() abstraction.BrokerTopic {
+	return boardPush.ID
+}
+
+type DownloadSuccess struct {
+	ID   abstraction.BrokerTopic // DSuccess
+	Data []byte
+}
+
+func (downloadSuccess DownloadSuccess) Topic() abstraction.BrokerTopic {
+	return downloadSuccess.ID
+}
```

Comment:
For these kinds of structures, as you are currently using them it won't work, instead do something like this:
```suggestion
	Data []byte
}

func (downloadSuccess DownloadSuccess) Topic() abstraction.BrokerTopic {
	return abstraction.BrokerTopic("blcu/download/success")
}
```
Do the same for the rest

Reply:
Ok, but the type conversion is redundant. I'll apply it without the conversion.